### PR TITLE
Resolves #15585 by allowing a name to be specified for the csrf token…

### LINKED ISSUE #15585: Custom CSRF Token cookie name not respected by IntegrationTestTrait
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.2.6 - 243a54d83438d9e9d5e3f4ccc1e379dcd20b5a38
* Platform and Target: PHP 7.4.9 CLI

### What you did
Within Application.php middleware I updated the CsrfProtectionMiddleware to use a custom cookie name, after this I reran my unit tests and found all POST requests were failing with 'Cake\Http\Exception\InvalidCsrfTokenException: Missing or incorrect CSRF cookie type.'.

Application.php
```
public function middleware(MiddlewareQueue $middlewareQueue): MiddlewareQueue
{
    ...

    // Cross Site Request Forgery (CSRF) Protection Middleware
    ->add(new CsrfProtectionMiddleware([
        'httponly' => true,
        'secure' => true,
        'cookieName' => '__Secure-myapp-csrfToken',
        'samesite' => 'Strict',
    ]))
```

ForgotControllerTest.php
```
class ForgotControllerTest extends IntegrationTestCase
{
    public function setUp(): void
    {
        parent::setUp();
        $this->disableErrorHandlerMiddleware();
        $this->enableCsrfToken();
        $this->enableSecurityToken();
    }

    public function testResetPassword()
    {
        $data = [
            'email' => 'test@test.com',
        ];
        $this->post('/forgot/reset-password/myguid', $data);
        $this->assertResponseOk();
    }
```

### What happened
Setting a custom cookie name within CsrfProtectionMiddleware is causing unit tests to fail with 'Cake\Http\Exception\InvalidCsrfTokenException: Missing or incorrect CSRF cookie type.'.

Using the default cookie name of 'csrfToken' allows the tests to pass as expected.

### What you expected to happen
I expected the IntegrationTestTrait to use the custom cookie name when setting tokens with the enableCsrfToken() function.

From my initial investigation I have found the IntegrationTestTrait.php is using a hardcoded cookie name on lines 658 - 671.
https://github.com/cakephp/cakephp/blob/95a6818297c8ea1b30d9013dae0cfbc57acdfc5d/src/TestSuite/IntegrationTestTrait.php#L658-L671

### PR DESCRIPTION
**Pull request by captain-redbeard**
Resolves https://github.com/cakephp/cakephp/issues/15585 by allowing a name to be specified for the CSRF cookie name.

### CODE REVIEW

File: src/TestSuite/IntegrationTestTrait.php [hunk 1]
```diff
@@ -237,11 +244,13 @@ public function setUnlockedFields(array $unlockedFields = []): void
      * Both the POST data and cookie will be populated when this option
      * is enabled. The default parameter names will be used.
      *
+     * @param string $cookieName The name of the csrf token cookie.
      * @return void
      */
-    public function enableCsrfToken(): void
+    public function enableCsrfToken(string $cookieName = 'csrfToken'): void
```
Comment (ADmad) [inline src/TestSuite/IntegrationTestTrait.php:250]: Given the change in method signature this needs to target `4.next` branch.

File: src/TestSuite/IntegrationTestTrait.php [hunk 2]
```diff
@@ -655,20 +664,20 @@ protected function _addTokens(string $url, array $data): array
...
-            if (!isset($this->_cookie['csrfToken']) && !isset($this->_session['csrfToken'])) {
+            if (!isset($this->_cookie[$this->_csrfCookieName]) && !isset($this->_session[$this->_csrfCookieName])) {
                 $token = $middleware->createToken();
-            } elseif (isset($this->_cookie['csrfToken'])) {
-                $token = $this->_cookie['csrfToken'];
+            } elseif (isset($this->_cookie[$this->_csrfCookieName])) {
+                $token = $this->_cookie[$this->_csrfCookieName];
             } else {
-                $token = $this->_session['csrfToken'];
+                $token = $this->_session[$this->_csrfCookieName];
```
Comment (markstory) [inline src/TestSuite/IntegrationTestTrait.php:672]: Given that we're using this property in the session as well, should it be `_csrfKeyName`?

File: src/TestSuite/IntegrationTestTrait.php (hunk 1 above)
Comment (captain-redbeard) [reply src/TestSuite/IntegrationTestTrait.php:250]: This has been updated to 4.next.

File: src/TestSuite/IntegrationTestTrait.php (hunk 2 above)
Comment (captain-redbeard) [reply src/TestSuite/IntegrationTestTrait.php]: Good point, this variable has been updated with the suggested name.

**Comment by othercorey** [comment]
@captain-redbeard Can you rebase this on latest 4.next? There are unrelated commits in the merge.

**Comment by markstory** [comment]
Thank you 👍

**Pull request merged**